serialized_data_interface/local_sdi.py

```diff
@@ -19,13 +20,12 @@ def create_new_metadata(metadata):
     return metadata
 
 
-def change_zip_file(charm_name, metadata):
-    temp_dir = f"{charm_name}.tmp"
-    charm_zip = f"{charm_name}.charm"
+def change_zip_file(charm_name, metadata, charm_path="."):
+    charm_zip = f"{charm_path}/{charm_name}.charm"
 
-    os.mkdir(temp_dir)
+    temp_dir = mkdtemp()
 
-    with ZipFileWithPermissions(f"{charm_name}.charm") as old_zip:
+    with ZipFileWithPermissions(charm_zip) as old_zip:
         old_zip.extractall(path=temp_dir)
 
     os.remove(charm_zip)
```

Comment:
This shouldn't be necessary, since passing `"w"` to `ZipFile` means that the old file will just be truncated before being written to

Reply:
What exactly are you referring to? the extract step?